packages/ui/src/composables/useTextColor.ts

```diff
@@ -1,19 +1,25 @@
 import { computed, getCurrentInstance, Ref, unref } from 'vue'
 import { useColors } from '../services/color-config/color-config'
 
-export const useTextColor = (componentColor: string | Ref<string>, isTransparent: boolean | Ref<boolean> = false) => {
-  const { props } = getCurrentInstance()!
+type PropsType = {
+  textColor: string,
+  color?: string
+}
+
+/**
+ * @param componentColor component color. By default `props.color`.
+ * @param isTransparent if transparent will return component color as text color.
+ * @returns Computed text color based on component's color if `props.textColor` if provided.
+ */
+export const useTextColor = (componentColor?: Ref<string> | undefined, isTransparent: boolean | Ref<boolean> = false) => {
```

Comment:
```suggestion
export const useTextColor = (componentColor?: Ref<string> | string | undefined, isTransparent: boolean | Ref<boolean> = false) => {
```

Reply:
I think, if we remove string, this will force use to use reactive variable. I don't believe we would have a component with static background.